src/library_webgpu.js

```diff
@@ -2536,6 +2538,39 @@ var LibraryWebGPU = {
     return adapter.features.size;
   },
 
+  wgpuAdapterGetInfo__deps: ['$stringToUTF8', '$lengthBytesUTF8', 'malloc', 'realloc'],
+  wgpuAdapterGetInfo: (adapterId, info) => {
+    var adapter = WebGPU.mgrAdapter.get(adapterId);
+    {{{ gpu.makeCheckDescriptor('info') }}}
+
+    function allocateUTF8String(stringPtr, stringValue) {
+      var stringSize = lengthBytesUTF8(stringValue) + 1;
+      if (!stringPtr) {
+        stringPtr = _malloc(stringSize);
+      } else {
+        _realloc(stringPtr, stringSize)
```

Comment:
The `wgpuAdapterGetInfo` function isn't supposed to do anything with incoming pointers, only overwrite them with new pointers. Using realloc will free the old pointers, which isn't supposed to happen.

Removing realloc I think makes this helper equivalent to `stringToNewUTF8`.

Reply:
Oh sorry, I see the incoming pointers are coming from `adapterWrapper` not from `info`.

Same point though - we're never supposed to implicitly free these allocations, the caller is responsible for doing that by calling `wgpuAdapterInfoFreeMembers`.

We should not need `infoPtrs` at all.

Reply:
Oh I didn't know `stringToNewUTF8` existed. It will simplify code for sure. Thanks!

Reply:
I just searched for the file `stringToUTF8` was defined in and found this, wasn't sure if you'd seen it because it is so similar to what you did :)

Reply:
And we're good now.